plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/KVMHAMonitor.java

```diff
@@ -87,41 +88,39 @@ protected void runHeartBeat() {
         synchronized (storagePool) {
```

Comment:
this hase bcome a very complex method. synchronised-for-if-try--if and then some after the closing catch. should it be disected for readability and maintainability?